cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -29,6 +29,110 @@
 namespace cudf {
 namespace detail {
 
+/**
+ * @brief Adds a pair of indices to the shared memory cache
+ *
+ * @param[in] first The first index in the pair
+ * @param[in] second The second index in the pair
+ * @param[in,out] current_idx_shared Pointer to shared index that determines
+ * where in the shared memory cache the pair will be written
+ * @param[in] warp_id The ID of the warp of the calling the thread
+ * @param[out] joined_shared_l Pointer to the shared memory cache for left indices
+ * @param[out] joined_shared_r Pointer to the shared memory cache for right indices
+ */
+__inline__ __device__ void add_pair_to_cache(size_type const first,
```

Comment:
This code was only used by conditional joins, so I moved it from `join_common_utils.cuh` to `conditional_join_kernels.cuh` where it is used. cuCollections implements similar cache/flush utilities that are used by hash and mixed joins but conditional joins do not use a hash-based algorithm so it's not immediately reusable.

Reply:
For the record, these used to be used by libcudf's hash join implementation before we migrated to cuco in https://github.com/rapidsai/cudf/pull/8934.